variables.tf

```diff
@@ -1,36 +1,267 @@
+    error_message = "Valid values 'retention_period' are: 7, 14, 30, 60, 90"
+  }
+}
+
+variable "data_storage" {
+  type = object({
+    logs_data = optional(object({
+      enabled              = optional(bool, false)
+      bucket_crn           = optional(string)
+      bucket_endpoint      = optional(string)
+      skip_cos_auth_policy = optional(bool, false)
+    }), {})
+    metrics_data = optional(object({
+      enabled              = optional(bool, false)
+      bucket_crn           = optional(string)
+      bucket_endpoint      = optional(string)
+      skip_cos_auth_policy = optional(bool, false)
+    }), {})
+    }
+  )
+  default = {
+    logs_data    = null,
+    metrics_data = null
+  }
+  validation {
+    condition     = (var.data_storage.logs_data.bucket_crn == null && var.data_storage.metrics_data.bucket_crn == null) || (var.data_storage.logs_data.bucket_crn != var.data_storage.metrics_data.bucket_crn)
+    error_message = "The same bucket cannot be used as both your data bucket and your metrics bucket."
+  }
+  validation {
+    error_message = "`bucket_crn` and `bucket_endpoint` must be included if logs_data `enabled` is true."
+    condition = (
+      lookup(var.data_storage.logs_data, "enabled", null) == null
+      ) || (
+      lookup(var.data_storage.logs_data, "enabled", false) == false
+      ) || (
+      lookup(var.data_storage.logs_data, "bucket_crn", null) != null &&
+      lookup(var.data_storage.logs_data, "bucket_endpoint", null) != null &&
+      lookup(var.data_storage.logs_data, "enabled", false) == true
+    )
+  }
+  validation {
+    error_message = "`bucket_crn` and `bucket_endpoint` must be included if metrics_data `enabled` is true."
+    condition = (
+      lookup(var.data_storage.metrics_data, "enabled", null) == null
+      ) || (
+      lookup(var.data_storage.metrics_data, "enabled", false) == false
+      ) || (
+      lookup(var.data_storage.metrics_data, "bucket_crn", null) != null &&
+      lookup(var.data_storage.metrics_data, "bucket_endpoint", null) != null &&
+      lookup(var.data_storage.metrics_data, "enabled", false) == true
+    )
+  }
+  description = "A logs data bucket and a metrics bucket in IBM Cloud Object Storage to store your IBM Cloud Logs data for long term storage, search, analysis and alerting."
+}
+
+variable "service_endpoints" {
+  description = "The type of the service endpoint that will be set for the IBM Cloud Logs instance. Allowed values: public-and-private."
+  type        = string
+  default     = "public-and-private"
+  validation {
+    condition     = contains(["public-and-private"], var.service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection. Allowed values: public-and-private."
+  }
+}
+
+##############################################################################
+# Event Notification
+##############################################################################
+
+variable "existing_en_instances" {
```

Comment:
As per our best practises defined in https://github.ibm.com/GoldenEye/issues/issues/12542 do not use short abbreviations like `en`. I know this is in the other repo now, but we should fix it here now before creating the first release of the new module. Suggest to name this `existing_event_notifications_instances`